schemas/client.yml

```diff
@@ -87,6 +103,38 @@
         simply taking the last label will not work well for effective TLDs such as "co.uk".
       example: co.uk
 
+    - name: subdomain
+      level: extended
+      type: keyword
+      short: The subdomain of the domain.
+      description: >
+        The subdomain is all of the labels under the registered_domain in a partially qualified
+        domain name, and all of the labels except the hostname in a fully qualified domain name.
+        
+        If the domain has multiple levels of subdomain, such as "sub2.sub1.example.com",
+        the subdomain field should contain "sub2.sub1", with no trailing period.
+      example: east
+
+    - name: hostname
+      level: core
+      type: keyword
+      short: Hostname of the client system.
+      description: >
+        Hostname of the client system.
+
+        It normally contains what the `hostname` command returns on the host machine, or the
+        host portion of a fully qualified domain name.
+      example: www
+
+    - name: country_code
+      level: extended
+      type: keyword
+      short: The country code portion of the domain.
+      description: >
+        The country code portion of the domain.
+
+      example: uk
```

Comment:
I'm not sure if we should go all the way to adding `country_code`.

I understand the value, but even just parsing out the eTLD and the registered_domain reliably puts a high burden on implementations, IMO. 

Asking them to also figure out the country code (only when appropriate, so not when it's a ".com", ".org" or ".beer") opens up other cans of worm: what about the region? Being from Quebec Canada I'm thinking of our beloved ".qc.ca" tld and the like.

@andrewkroh What do you think? Does the public suffix list help fill this country code (aka it's not much more of a burden)? Or does it not help with country code?

If we decide to keep it, we should at least say in the description that it should be filled only when there's a country code :-)

Reply:
I feel like parsing issues shouldn't be a consideration in terms of the availability of accurate sub fields.   I'll update the description to say it should only be filled in when there is a country code.

Reply:
The country code TLDs are a subset of all TLDs. And we already have a `top_level_domain` field so this will be duplicated. If is useful to know that a TLD is a country code then maybe that's a boolean flag like `is_country_code` to indicate the `top_level_domain` is a country.

Similarly I had thought other attributes of the domain name could be flags. Like whether it contains puny codes with `is_internationalized` (https://github.com/elastic/beats/issues/5809#issuecomment-451710649).

Reply:
@andrewkroh that leaves me wondering 2 things (and I'm serious... not just pondering to be annoying, which I may have been known to do on occasion :) 

1) where do we draw the line for cutting up aggregate things (like a fqdn) into its component parts?

tld is part of a fqdn - and we duplicates it, but cc (and apparently ?region code? thanks for that Quebec!) are also part of a  tld...  should we be cutting a fqdn up into all of its component parts? or just the major ones? what criteria should we use for "major"? 

I can't see visualizing cc against tld, but I could potentially see utilizing cc as a feature, or as an influencer / bucket for analysis (region code should result in the creators immediate relegation to their own personal purgatory as far as I am concerned :) 

2) I love the "is_internationalized" flag idea, and think theres probably some others we could consider as atomic features (digit ratio, number of special characters, entropy, levenstein distance to similar popular domains, subdomain count, subdomain length, etc.) but, well where do we draw the line with that too?

Reply:
added is_internationalized, removed country_code and added to the tld description ti indicate its the spot to tld/cctld